website/content/en/docs/upgrading-sdk-version/v1.5.0.md

```diff
@@ -0,0 +1,110 @@
+Ensure that the PROJECT file results are like:
+```yml domain: example.com projectName: memcached-operator resources: - api:
+    crdVersion: v1
+    namespaced: true
+  domain: example.com
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+version: "3" ```
+
+_See [#4402](https://github.com/operator-framework/operator-sdk/pull/4402) for more details._
+
+## PROJECT config version 3-alpha must be upgraded to 3
+
+PROJECT config version 3-alpha has been stabilized as [version 3](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/migration/manually_migration_guide_v2_v3.md) (the `version` key in your PROJECT file), and contains a set of config fields sufficient to fully describe a project. While this change is not technically breaking because the spec at that version was alpha, it was used by default in `operator-sdk` commands so should be marked as breaking and have a convenient migration path. The `alpha config-3alpha-to-3` command will convert most of your PROJECT file from version 3-alpha to 3, and leave comments with directions where automatic conversion is not possible:
+```console $ cat PROJECT version: 3-alpha resources: - crdVersion: v1 ... $ operator-sdk alpha config-3alpha-to-3 Your PROJECT config file has been converted from version 3-alpha to 3. Please make sure all config data is correct. $ cat PROJECT version: "3" resources: - api:
+    crdVersion: v1
+... ```
+
+_See [#4613](https://github.com/operator-framework/operator-sdk/pull/4613) for more details._
+
+## (go/v3) Upgrade controller-runtime to v0.7.2
+
+In your go.mod file, upgrade `sigs.k8s.io/controller-runtime` to v0.7.2.
+
+_See [#4626](https://github.com/operator-framework/operator-sdk/pull/4626) for more details._
+
+## (go/v3) Add a `system:controller-manager` ServiceAccount to your project.
+
+A non-default ServiceAccount `controller-manager` is scaffolded on `operator-sdk init`, to improve security for operators installed in shared namespaces. To add this ServiceAccount to your project, do the following:
+```sh # Create the ServiceAccount. cat <<EOF > config/rbac/service_account.yaml apiVersion: v1 kind: ServiceAccount metadata:
+  name: controller-manager
+  namespace: system
+EOF
+# Add it to the list of RBAC resources. echo "- service_account.yaml" >> config/rbac/kustomization.yaml
+# Update all RoleBinding and ClusterRoleBinding subjects that reference the operator's ServiceAccount. find config/rbac -name *_binding.yaml -exec sed -i -E 's/  name: default/  name: controller-manager/g' {} \;
+# Add the ServiceAccount name to the manager Deployment's spec.template.spec.serviceAccountName. sed -i -E 's/([ ]+)(terminationGracePeriodSeconds:)/\1serviceAccountName: controller-manager\n\1\2/g' config/manager/manager.yaml ```
```

Comment:
````suggestion
```sh
# Create the ServiceAccount.
cat <<EOF > config/rbac/service_account.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: controller-manager
  namespace: system
EOF

# Add it to the list of RBAC resources.
echo "- service_account.yaml" >> config/rbac/kustomization.yaml

# Update all RoleBinding and ClusterRoleBinding subjects that reference the operator's ServiceAccount.
find config/rbac -name *_binding.yaml -exec sed -i -E 's/  name: default/  name: controller-manager/g' {} \;

# Add the ServiceAccount name to the manager Deployment's spec.template.spec.serviceAccountName.
sed -i -E 's/([ ]+)(terminationGracePeriodSeconds:)/\1serviceAccountName: controller-manager\n\1\2/g' config/manager/manager.yaml
```
````